src/mozanalysis/sizing.py

```diff
@@ -226,10 +248,26 @@ def get_single_window_data(
             )
         )
 
-        return bq_context.run_query(
+        output = bq_context.run_query(
             self._metrics_sql, full_res_table_name, replace_tables
         ).to_dataframe()
 
+        if (len(metric_list_sources) < len(metric_list)) or (
+            len(target_list_sources) < len(target_list)
+        ):
+            # case where there is a custom definition
+            # so sources have not been checked
+            for metric_obj in metric_list:
+                if all(output[metric_obj.name] == 0):
+                    raise warnings.warn(
+                        (
+                            f"Metric {metric_obj.name} is all 0, which may indicate"
+                            + " segements and metric do not have a common source"
```

Comment:
I like this check, and I think it should always run -- for example, we could have two Firefox Desktop metrics, but one uses a Glean client_id and one uses a legacy client_id, and this check would fail.

Reply:
Fixed